api/src/main/java/org/openmrs/api/context/Context.java

```diff
@@ -239,7 +239,7 @@ public static UserContext getUserContext() {
 	 *
 	 * @return the current ServiceContext
 	 */
-	static ServiceContext getServiceContext() {
+	static synchronized ServiceContext getServiceContext() {
```

Comment:
I can see that the method body was already synchronised. Can you instead remove the outer if (serviceContext == null)? Because i see it duplicated in the inner block.